### DISCUSSION TITLE
recommended ways to handle contexts with new reactive primitives?

### DISCUSSION THREAD

**Discussion by lictex** [sycamore-rs/sycamore#372] [Q&A] [answered]
they re working well in most cases really
currently context types are required to be `'static` (by `Any`) so the pattern below wont working
```rust
pub struct Contexts<'l> {
    pub ctx_0: &'l Signal<i32>,
    pub ctx_1: &'l Signal<i32>,
    // ...
}
```
the choices now are using `RcSignal` (many `clone()`s again..), or wrapping the whole struct in one signal (less performant? not tested tho)
but in theory this limit seems could be removed imo as contexts are sharing lifetime with the scope, not really `'static`

---

got a solution for my project with [rrevenantt/better_any](https://github.com/rrevenantt/better_any) but far from elegant. still looking for if theres a better way...

**Comment by lukechu10** [comment, marked as answer]
Hi there sorry for taking so long to answer. The recommended pattern is to use `RcSignal`. Note that the `clone()`s are not actually necessary because `use_context()` returns a reference so you can just hold a `&RcSignal` which is `Copy`.

**Reply by lictex** [reply]
thanks for the help!
...i mistakenly thought `RcSignal` should always be used with `clone()` like older versions of sycamore
